src/arithmetic/canonical.cc

```diff
@@ -532,18 +534,22 @@ class Canonical::Internal : public IRMutator {
       return make_zero(t);
     }
   }
+public:
+  Map<Var, Range> vrange_;
```

Comment:
I'm not sure what's the most appropriate way is to pass variable range information into Canonical::Internal.
One alternative is to define vrange_ as private and define a setter/getter methods for it.

Reply:
public field is fine as it is already internal class

Reply:
good point. but it may become a slippery slope. I'll take your suggestion and leave it as it is for now.